process/preprocess/eventsPreProcessorV0.go

```diff
@@ -67,44 +67,24 @@ func (d *eventsPreProcessorV0) SaveBlock(marshalledData []byte) error {
 	return nil
 }
 
-func (d *eventsPreProcessorV0) setTransactionsPool(txsPool *data.TransactionsPool) (*outport.TransactionPool, error) {
+func (d *eventsPreProcessorV0) parseTransactionsPool(txsPool *data.TransactionsPool) (*outport.TransactionPool, error) {
 	if txsPool == nil {
 		return nil, process.ErrNilTransactionsPool
 	}
 
 	txs := make(map[string]*outport.TxInfo)
-
 	if txsPool.Txs != nil {
-		for hash, txHandler := range txsPool.Txs {
-			txs[hash] = &outport.TxInfo{
-				Transaction:    txHandler.TransactionHandler,
-				FeeInfo:        &txHandler.FeeInfo,
-				ExecutionOrder: uint32(txHandler.ExecutionOrder),
-			}
-		}
+		txs = d.parseTxs(txsPool.Txs)
```

Comment:
why not move the if ... != nil checks inside the newly added methods?

this way, this method will be:

```go
func (d *eventsPreProcessorV0) parseTransactionsPool(txsPool *data.TransactionsPool) (*outport.TransactionPool, error) {
	if txsPool == nil {
		return nil, process.ErrNilTransactionsPool
	}

	return &outport.TransactionPool{
		Transactions:         d.parseTxs(txsPool.Txs),
		SmartContractResults: d.parseScrs(txsPool.Scrs),
		Logs:                 d.parseLogs(txsPool.Logs),
	}, nil
}
```